app/controllers/mixins/containers_common_mixin.rb

```diff
@@ -79,6 +81,20 @@ def show_container(record, controller_name, display_name)
                       :url  => "/#{controller_name}/show/#{record.id}" \
                                "?display=#{@display}&refresh=n")
       perf_gen_init_options # Intialize options, charts are generated async
+    elsif @display == "compliance_history"
+      count = params[:count] ? params[:count].to_i : 10
+      session[:ch_tree] = compliance_history_tree(record, count).to_json
+      session[:tree_name] = "ch_tree"
+      session[:squash_open] = (count == 1)
+      if count == 1
+        drop_breadcrumb(:name => _("%{name} (Latest Compliance Check)") % {:name => record.name},
+                        :url  => "/#{controller_name}/show/#{record.id}?display=#{@display}&refresh=n")
+      else
+        drop_breadcrumb(:name => _("%{name} (Compliance History - Last %{number} Checks)") %
```

Comment:
Is it possible to change this to something like:

```ruby
elsif @display == "compliance_history"
  # set local variables
  update_session_for_compliance_history(pass any local variables we need)
  drop_breadcrumb_for_compliance_history(record.name..)
```

It doesn't have to be here but it would be nice to start creating methods for future re-use/refactoring, especially the drop_breadcrumb... it looks like most of the elsifs do this.

Honestly, I don't know what these conditionals really do but they seems to be doing similar things in terms of the breadcrumbs.